Algorithms/Numeric/AutomorphicNumber.cs

```diff
@@ -0,0 +1,84 @@
+using System;
+using System.Collections.Generic;
+using System.Numerics;
+
+namespace Algorithms.Numeric
+{
+    /// <summary>
+    /// Calculates Automorphic numbers. A number is said to be an Automorphic number
+    /// if the square of the number will contain the number itself at the end.
+    /// </summary>
+    public static class AutomorphicNumber
+    {
+        /// <summary>
+        /// Generates a list of automorphic numbers that are between <paramref name="lowerBound"/> and <paramref name="upperBound"/>
+        /// inclusive.
+        /// </summary>
+        /// <param name="lowerBound">The lower bound of the list.</param>
+        /// <param name="upperBound">The upper bound of the list.</param>
+        /// <returns>A list that contains all of the automorphic numbers between <paramref name="lowerBound"/> and <paramref name="upperBound"/> inclusive.</returns>
+        /// <exception cref="ArgumentException">If the <paramref name="lowerBound"/> or <paramref name="upperBound"/> is not greater than zero.</exception>
+        /// <remarks>If the <paramref name="lowerBound"/> is greater than the <paramref name="upperBound"/> then the two bounds are flipped.</remarks>
+        public static List<long> GetAutomorphicNumbers(long lowerBound, long upperBound)
+        {
+            if (lowerBound < 1)
+            {
+                throw new ArgumentException($"Lower Bound must be greater than 0: Actual value {lowerBound}");
+            }
+
+            if (upperBound < 1)
+            {
+                throw new ArgumentException($"Upper Bound must be greater than 0: Actual value {upperBound}");
+            }
+
+            if (lowerBound > upperBound)
+            {
+                var tmp = lowerBound;
+                lowerBound = upperBound;
+                upperBound = tmp;
+            }
```

Comment:
I think this should throw an error instead. Being defensive can lead to unexpected results and errors in application code. See https://en.wikipedia.org/wiki/Offensive_programming

Reply:
Didn't think about that, thanks for the insight!
Changed it to throw an ArgumentException instead.